scripts/abins/input/tester.py

```diff
@@ -206,21 +215,30 @@ def check(self, *, name: str, loader: AbInitioLoader, extension: str = None, max
             max_displacement_kpt=max_displacement_kpt,
         )
 
-    def _read_ab_initio(self, loader=None, filename=None, extension=None):
+    def _read_ab_initio(self, loader=None, filename=None, extension=None, **loader_kwargs) -> Dict[str, Any]:
         """
         Reads data from .{extension} file.
-        :param loader: ab initio loader
-        :param filename: name of file with vibrational or phonon data (name + extension)
-        :returns: vibrational or phonon data
+
+        Args:
+            loader: ab initio loader class (to instantiate and test)
+            filename: prefix of file with vibrational or phonon data
+            extension: extension of data file (e.g. ".phonon")
+            **loader_kwargs: passed to loader.__init__()
+
+        returns: vibrational or phonon data
+
         """
         # 1) Read data
         try:
             read_filename = abins.test_helpers.find_file(filename=filename + "." + extension)
-            ab_initio_reader = loader(input_ab_initio_filename=read_filename)
         except ValueError:
+            read_filename = ""
+        if not read_filename:
             read_filename = abins.test_helpers.find_file(filename=filename + "." + extension.upper())
```

Comment:
if a `ValueError` was procured to deal with line:
```python
read_filename = abins.test_helpers.find_file(filename=filename + "." + extension)
```
then perhaps one could procure another `ValueError` to deal with line
```python
read_filename = abins.test_helpers.find_file(filename=filename + "." + extension.upper())
```
?

Reply:
I think the goal here was to avoid a confusing nested try-except, and always hit the same error call. But I the variable usage here is also a bit confusing.

This pattern is a bit ugly either way and appears in a few places in this file. Part of the answer might be to make extension-case-insensitivity a feature of `abins.test_helpers.find_file`; we can actually dodge all the try-except logic by looking at the filename before passing it to to the load function.

Reply:
Ok, that looks better to me!